src/App.tsx

```diff
@@ -1,14 +1,111 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos, getUser } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+
+export type CompletedFilter = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>([]);
+  const [loadingTodos, setLoadingTodos] = useState<boolean>(false);
+  const [loadingUser, setLoadingUser] = useState<boolean>(false);
+  const [error, setError] = useState<boolean>(false);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [completedFilters, setCompletedFilters] = useState<CompletedFilter>('all');
+  const [inputFilter, setInputfilter] = useState<string>('');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[] | null>(todos);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setLoadingTodos(true);
+      setError(false);
+      setUser(null);
+
+      getTodos().then(setTodos).catch(e => {
+        setError(e.message);
+      }).finally(() => setLoadingTodos(false));
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    const loadUser = async () => {
+      setLoadingUser(true);
+      setError(false);
+      setUser(null);
+
+      if (todo) {
+        getUser(todo?.userId).then(setUser).catch(e => {
+          setError(e.message);
+        }).finally(() => setLoadingUser(false));
+      }
+    };
+
+    loadUser();
+  }, [todo]);
```

Comment:
since user is used only in TodoModal, I think you can fetch it in that component